test/blackbox-tests/test-cases/github25/run.t

```diff
@@ -20,5 +20,5 @@ We need ocamlfind to run this test
   File ".../plop/META", line 1, characters 0-0:
   Error: Library "une-lib-qui-nexiste-pas" not found.
   -> required by library "plop.ca-marche-pas" in .../plop
-  Hint: try: dune external-lib-deps --missing --display short --only pas-de-bol @install
+  Hint: try: dune external-lib-deps --missing --only pas-de-bol --display short @install
```

Comment:
It's a bit strange that this changes the output. Do you know why that is the case?

Reply:
Arguments are collected into a set and the compare function of this set compares ids. From cmdliner_info.ml:

```ocaml
module Arg = struct
  type t = arg
  let compare a0 a1 = (compare : int -> int -> int) a0.id a1.id
end

module Args = Set.Make (Arg)
```

This PR changes a bit the shape of the command line description, from `const f $ arg1 $ arg2 $ arg3` to `const f $ ((and+ arg1 arg2) arg3)`. Given that `arg{1,2,3}` are not evaluated in the same order anymore, the identifiers are not generated in the same order as well and this changes the order of arguments in the set.

Reply:
There is not much we can do for now without changing cmdliner first.

Reply:
Thanks for the explanation. Is this guaranteed to be evaluated in the same order as the native version in the compiler? Otherwise, would it make sense to fix the order of evaluation by emitting `let v1 = arg1 in let v2 = arg2 in and+ v1 v2`?

Reply:
We try to avoid these kinds of differences in behavior between native and bytecode, however there is no strong guarantee that they match. Regarding the rewriting, we should match what 4.08 does. After playing with the toplevel, it does seem that  we should generate these bindings:

```ocaml
# let (let+) x f = f x;;
val ( let+ ) : 'a -> ('a -> 'b) -> 'b = <fun>
# let (and+) a b = a, b;;
val ( and+ ) : 'a -> 'b -> 'a * 'b = <fun>
# let+ () = print_endline "a" and+ () = print_endline "b" and+ () = print_endline "c" in ();;
a
b
c
- : unit = ()
# (let+) 1 ignore;;
- : unit = ()
# (let+) ((and+) ((and+) (print_endline "a") (print_endline "b")) (print_endline "c")) (fun (((), ()), ()) -> ());;
c
b
a
```